src/schema-helpers/middleware-setters.ts

```diff
@@ -10,22 +11,28 @@ import {
     softRemoveFunc,
 } from './middleware-functions';
 
-export const addQueryMiddleware = (schema: Schema, headers: PolarisRequestHeaders) => {
+export const addQueryMiddleware = (
+    schema: Schema,
+    headers: PolarisRequestHeaders,
+    modelConfiguration: ModelConfiguration,
+) => {
     const findHandlerFunc = getFindHandler(headers);
     ['find', 'findOne', 'findOneAndUpdate', 'update', 'count', 'updateOne', 'updateMany'].forEach(
         middleware => {
             schema.pre(middleware, findHandlerFunc as any);
         },
     );
     schema.pre('aggregate', preAggregate);
-    schema.statics = {
-        ...schema.statics,
-        remove: softRemoveFunc,
-        deleteOne: singleSoftRemove,
-        deleteMany: softRemoveFunc,
-        findOneAndDelete: findOneAndSoftDelete,
-        findOneAndRemove: findOneAndSoftDelete,
-    };
+    if (modelConfiguration.allowSoftDelete) {
+        schema.statics = {
+            ...schema.statics,
+            remove: softRemoveFunc,
+            deleteOne: singleSoftRemove,
```

Comment:
rename softRemoveFunc, singleSoftRemove - not good names
maybe call it softRemove and softRemoveOne / softDeleteOne